crates/bevy_app/src/app.rs

```diff
@@ -1078,11 +1139,22 @@ mod tests {
 
     #[test]
     #[should_panic]
-    fn test_is_plugin_added_works_during_finish() {
+    fn plugin_cannot_be_added_during_finish() {
         let mut app = App::new();
         app.add_plugins(PluginA);
+        assert_eq!(app.plugins_state(), PluginRegistryState::Init);
+
+        app.update_plugins(); // Build
+        assert_eq!(app.plugins_state(), PluginRegistryState::Building);
+        app.update_plugins(); //
+        assert_eq!(app.plugins_state(), PluginRegistryState::Finalizing);
+        app.update_plugins();
+        assert_eq!(app.plugins_state(), PluginRegistryState::Done);
+
         app.add_plugins(PluginE);
-        app.finish();
+
+        app.update_plugins();
+        // app.finish();
```

Comment:
Should this be commented out? Its purpose isn't clear when reading the test.

Reply:
I removed this, it's not needed for the test.